### PR TITLE
Spring boot online book store add jwt

### PR DESCRIPTION
**Pull request by snowfallen**

### CODE REVIEW

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest httpServletRequest) {
+        String bearerToken = httpServletRequest.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer")) {
+            return bearerToken.substring(7);
```
Comment (okuzan) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]: make Bearer a constant, or import existing one, Spring should have it, then 7 would be .length of it - they are logically connected, and good code should make it obvious :)

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest httpServletRequest) {
+        String bearerToken = httpServletRequest.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]: Use Spring constant HttpHeaders.AUTHORIZATION

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String TOKEN_PREFIX = "Bearer";
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String TOKEN_PREFIX = "Bearer ";
```

File: src/main/java/book/store/config/SecurityConfig.java [hunk 4]
```diff
@@ -32,18 +37,25 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
         return http
                 .cors(AbstractHttpConfigurer::disable)
                 .csrf(AbstractHttpConfigurer::disable)
-                .authorizeHttpRequests(
-                        auth -> auth
-                                .requestMatchers("/auth/**", "/swagger-ui/**", "/v3/api-docs/**")
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/config/SecurityConfig.java:37]: Why did you decide to remove "/swagger-ui/**", "/v3/api-docs/**" ?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**